### PR TITLE
`dl-v2` handling graceful termination in license deployment

### PR DESCRIPTION
**Pull request by aniketsinghrawat**

### CODE REVIEW

File: weather_dl_v2/license_deployment/fetch.py [hunk 1]
```diff
@@ -148,14 +148,20 @@ def fetch_request_from_db():
...
         log_count = 0
         while True:
-            # Fetch a request from the database
-            request = fetch_request_from_db()
+            # Check if sigterm was recived for gracefull termination
+            if not killer.kill_now:
+                # Fetch a request from the database.
+                request = fetch_request_from_db()
+            else:
+                logger.warning('SIGTERM recieved. Stopping futher requets processing.')
```
Comment (mahrsee1997) [inline weather_dl_v2/license_deployment/fetch.py:163]: typo: further.

Comment (aniketsinghrawat) [reply weather_dl_v2/license_deployment/fetch.py]: done

File: weather_dl_v2/license_deployment/fetch.py [hunk 2]
```diff
@@ -148,14 +148,20 @@ def fetch_request_from_db():
...
+    killer = GracefulKiller()
     with ThreadPoolExecutor(concurrency_limit) as executor:
         # Disclaimer: A license will pick always pick concurrency_limit + 1
         # parition. One extra parition will be kept in threadpool task queue.
 
         log_count = 0
         while True:
-            # Fetch a request from the database
-            request = fetch_request_from_db()
+            # Check if SIGTERM was recived for gracefull termination.
```
Comment (mahrsee1997) [inline weather_dl_v2/license_deployment/fetch.py:158]: nit: graceful

**Review by mahrsee1997 — APPROVED** [review]
LGTM!

**Pull request merged**